From 3d350b741186740a28eb893e253056d75635d060 Mon Sep 17 00:00:00 2001
From: Archit Taneja <archit@ti.com>
Date: Fri, 25 Oct 2013 20:23:20 +0530
Subject: [PATCH 0749/1115] omapdss: dss_dpll: Fix bugs in DSS_PLL_CONTROL
 configuration

DSS_PLL_CONTROL configures muxes which chooses between the VIDEO1/2 and HDMI
PLLs for each LCD manager. There were some mistakes in the bitfield locations
and values of the registers. Fix those accordingt to TRM.

Signed-off-by: Archit Taneja <archit@ti.com>
(cherry picked from commit dce7bf844b024906580606f8fde18d220e74df0f)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/video/omap2/dss/dss_dpll.c |   18 +++++++++---------
 1 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/video/omap2/dss/dss_dpll.c b/drivers/video/omap2/dss/dss_dpll.c
index 170c8fa..71ed545 100644
--- a/drivers/video/omap2/dss/dss_dpll.c
+++ b/drivers/video/omap2/dss/dss_dpll.c
@@ -404,7 +404,7 @@ void dss_dpll_set_control_mux(enum omap_channel channel, enum dss_dpll dpll)
 			val = 0;
 			break;
 		default:
-			printk("error in mux config\n");
+			DSSERR("error in mux config for LCD\n");
 			return;
 		}
 	} else if (channel == OMAP_DSS_CHANNEL_LCD2) {
@@ -413,28 +413,28 @@ void dss_dpll_set_control_mux(enum omap_channel channel, enum dss_dpll dpll)
 
 		switch (dpll) {
 		case DSS_DPLL_VIDEO1:
-			val = 1;
+			val = 0;
 			break;
 		case DSS_DPLL_VIDEO2:
-			val = 0;
+			val = 1;
 			break;
 		default:
-			printk("error in mux config\n");
+			DSSERR("error in mux config for LCD2\n");
 			return;
 		}
 	} else {
-		start = 7;
-		end = 8;
+		start = 8;
+		end = 7;
 
 		switch (dpll) {
 		case DSS_DPLL_VIDEO1:
-			val = 0;
+			val = 1;
 			break;
 		case DSS_DPLL_VIDEO2:
-			val = 1;
+			val = 0;
 			break;
 		default:
-			printk("error in mux config\n");
+			DSSERR("error in mux config for LCD3\n");
 			return;
 		}
 	}
-- 
1.7.5.4

